core/src/driver/scheduler.rs

```diff
@@ -94,12 +93,12 @@ pub enum SchedulerKind {
 
 impl SchedulerKind {
     /// Creates a new scheduler based on the parameters.
-    pub fn create<'a>(
+    pub fn create(
         &self,
-        exchange: &'a dyn StableXContract,
-        driver: &'a (dyn StableXDriver + Sync),
+        exchange: Arc<dyn StableXContract + Send + Sync>,
+        driver: Arc<dyn StableXDriver + Send + Sync>,
```

Comment:
Maybe all our components that need to be shared across threads should be `trait Component: Send + Sync` because we need them to be like this pretty much everywhere.